### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by maddygriz**

### CODE REVIEW

**Review by NicoleEO — CHANGES_REQUESTED** [review]
xmlParsing.R comments

- [x] throughout code roxygen tag @importFrom, package name should be before function names
- [x] Change "some AOIs do no" --> "some AOIs do not"
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/174cfe7ddba836e40d987a7a1d7cfe434ab6b00a/R/xmlParsing.R#L143
- [x] Do you even need the column rename? Also, is the warning saying the opposite of what's going on? There are more annotations then AOIs. So some AOIs in annotation file not in experiment?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/174cfe7ddba836e40d987a7a1d7cfe434ab6b00a/R/xmlParsing.R#L142

Column rename allows GxT object to go through same logic as the lab worksheet. This package only works on 1 slide at a time but someone could give an entire study (multiple slides) worth of annotations. 

- [x] Title case is usually reserved for classes (PhysicalSizeX and Y, Fluorescence)
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/xmlParsing.R#L168
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/xmlParsing.R#L207
- [x] Put globals at the top of the file
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/xmlParsing.R#L180
- [x] Didn't chan just get 1 added in Line 232? The line quoted below also has a +1 so you are skipping a line for the ==ChannelInfo check. Maybe this should just be rewritten to for loop?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/xmlParsing.R#L244
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/xmlParsing.R#L255

I agree this is not the best written code but I don't think a for loop will work. The XML files have varying lines for the fluorescence data and this was actually the easiest way I could figure out how to accurately get the data. I will look into condensing this code if I have time before the release but it's probably at the bottom of my list. 

@maddygriz just to confirm so that's correct you add one (L232) then and another (L244/L255) before using to do check for Channel Info? So 2 indices away from starting point.
@NicoleEO yes the add on L232 gets to the correct line and then lines L244/255 looks at the next line to see what the next line looks like to affect the logic. 

- [x] What software is being referenced here?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/xmlParsing.R#L326

No software, literally the number of masks in an ROI. But it is "calculated" from the lab worksheet.

**Review by NicoleEO — CHANGES_REQUESTED** [review]
utils.R comments

- Kind of odd that we now have two functions for read lab worksheet. It would be good in the future to importFrom GeomxTools. Can you start an issue documenting all v2 suggestsions?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/utils.R#L31
- Add these as globals in the beginning of file or as function parameters with these set as defaults. 
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/utils.R#L82
- Why is this redefined?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/utils.R#L98
- Use variable and basename() instead of real path name
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/utils.R#L105
- Add rox comments for these imports so we don't lose track in the future
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/utils.R#L112

**Comment by NicoleEO** [comment]
removeSamples.R looks good, we should consider adding accessor functions in v2 to stop using direct slot access

**Comment by NicoleEO** [comment]
readSpatialOverlay.R good

**Comment by NicoleEO** [comment]
plottingBasics.R
- [ ] Ellipses not used anywhere in this function. https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/plottingBasics.R#L60
- [ ] If it is hard assigned and not a user input do you need the toupper and if stmt? Also, use the <- assignment operator for axis = X
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/plottingBasics.R#L242-L248
- [ ] Do you need this if? X is axis and the calc is exactly the same in if and else
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/plottingBasics.R#L275
- [ ] You have ellipses passed to both of these args. Do all args for one work for the other? Or is some parsing required? If you don't want to fix now suggest adding comment to do one or the other for the time being and add to v2 update.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/plottingBasics.R#L281
- [ ] try to use intL for numbers throughout. You can fix later not urgent.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/plottingBasics.R#L464

**Comment by NicoleEO** [comment]
omeExtraction

- [ ] This could cause some issues depending on space in working directory. Might be nice to use a required param for directory to save to.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/omeExtraction.R#L21
- [ ] Is this an imported function? If so, add importFrom.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/omeExtraction.R#L30
- [ ] same here
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/omeExtraction.R#L46
- [ ] same about saveFile, add path param
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/omeExtraction.R#L83

**Comment by NicoleEO** [comment]
- [ ] 1L
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L101
- [ ] is this right you potentially could calc maxheight - y twice if initial ymax > ymin? Done in L338 and again in L360 through 364
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L338
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L360-L364
- [ ] Is this an imported function? Add importFrom.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L455
- [ ] Does this run the risk of being a different dimension than the list of AOIs in the object because you could filter out certain AOIs right with the logic statements? Or are all the samples kept in the overlay object and only coords used for plotting samples?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L357-L365
- [ ] v2 enhancement, add list of samples ignored with warning
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L413
- [ ] is boundary imported, if so importFrom? I couldn't find where this was defined by search, but I feel like this is one of your funcs.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L527
- [ ] Add importFrom
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/imageManipulation.R#L539

**Comment by NicoleEO** [comment]
coordinateGeneration.R

- [ ] Can we put x before y, just to match the order you have throughout?
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/coordinateGeneration.R#L142
- [ ] 1L
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/coordinateGeneration.R#L167

**Comment by NicoleEO** [comment]
addPlottingFactor.R
- [ ] Add importFrom
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/addPlottingFactor.R#L128

**Comment by NicoleEO** [comment]
addImage is good

**Comment by NicoleEO** [comment]
SpatialPosition looks good

**Comment by NicoleEO** [comment]
-[ ] What is the signature these methods are matching on? Looks like these constructors don't really do anything. You can delete these and the generic probably and just keep the default constructor in prototype.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/SpatialOverlay-class.R#L62-L93
- [ ] rox comments and exports of all the accessors. I found them manually entered. We should add to v2 to get this in rox comments instead. I have to do this in GxT as well at some point.
https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/blob/fd53a5690acd404c133be82a2499975a4e4d098d/R/SpatialOverlay-class.R#L154

**Comment by maddygriz** [comment]
image

File: R/addImage.R [hunk 1]
```diff
@@ -120,19 +120,19 @@ addImageFile <- function(overlay, imageFile = NULL, res = NULL){
...
-    if(is.null(overlay@image$resolution)){
+    if(is.null(res(overlay))){
         difRes <- -1
     }else{
-        difRes <- res - overlay@image$resolution
+        difRes <- res - res(overlay)
     }
     
-    overlay@image <- list(filePath = ometiff, 
+    overlay@image  <- list(filePath = ometiff,
```
Comment (NicoleEO) [inline R/addImage.R:135]: Looks like you accidentally added an extra space before <-

File: R/SpatialOverlay-class.R [hunk 2]
```diff
@@ -227,3 +213,18 @@ setGeneric("res", signature = "object",
            function(object) standardGeneric("res"))
 setMethod("res", "SpatialOverlay", function(object) 
     object@image$resolution)
+
```
Comment (NicoleEO) [inline R/SpatialOverlay-class.R:216]: Add to export generics and methods these 3 functions

Comment (NicoleEO) [inline R/SpatialOverlay-class.R:216]: Add to export generics and methods these 3 functions

**Review by NicoleEO — CHANGES_REQUESTED** [review]

**Comment by NicoleEO** [comment]
@maddygriz also see my mentions (@maddygriz) inside my previous comments for a couple of questions.

**Comment by maddygriz** [comment]
no tests were written for adding the RBioFormat functions to the code base. The saved snaps used in vdiffr were made before changing the location of the code so if the generated images are the same between code bases, implementation should be the same.

**Comment by maddygriz** [comment]
image
image

- one warning was taken care of 
image

- R CMD build errors seem to be based on latex, will look into them
image

**Comment by NicoleEO** [comment]
Comments from here on are regarding [latest updates](https://github.com/Nanostring-Biostats/SpatialOmicsOverlay/compare/2ef56fb998d6083c1e969556cf73498049d02177...main#diff-9aadfeb2619230f5074f5536c75faeae15e69bef5c2706568727dfc96fb9e7a0L45).

**Comment by NicoleEO** [comment]
@maddygriz If only very minor edits result from the below comments, let's wait until you get a reviewer assigned to fix these so hold the PR into main until then. In the meantime, let's get a dev branch setup.
- [ ] Line 45 and on in addImage, is difRes == 0 just not possible or you don't care about that case for running the lines that previously were in that case.
previously had: if(difRes != 0) 
- [ ] L523 in imageManipulation, I believe I saw this if statement a few times. Sometime with a ! and sometimes without. You may want to search and make sure coords is set right in your ifs throughout.
image
- [ ] Line 416 in tests/testthat/test_plotting.R add another line at the bottom of script
- [ ] In RBioFormatsFunctions.R, I found a few instances of = instead of -> can you double-check to replace these?

**Pull request merged**